glue-codes/openfast-cpp/src/OpenFAST.H

```diff
@@ -182,75 +329,134 @@ class OpenFAST {
+  //! Set the velocity at velocity node 'iNode' for turbine number 'iTurbGlob'
   void setVelocity(std::vector<double> & velocity, int iNode, int iTurbGlob);
+  //! Set the velocity at force node 'iNode' for turbine number 'iTurbGlob'
   void setVelocityForceNode(std::vector<double> & velocity, int iNode, int iTurbGlob);
+  //! Map the velocity from the force nodes to the velocity nodes using linear interpolation along each blade and the tower.
   void interpolateVel_ForceToVelNodes();
+  //! Get the node type (HUB, BLADE, TOWER) of force node number 'iNode' for turbine number 'iTurbGlob'
   ActuatorNodeType getForceNodeType(int iTurbGlob, int iNode);
-  void getForceNodeCoordinates(std::vector<double> & currentCoords, int iNode, int iTurbGlob);
-  void getForceNodeOrientation(std::vector<double> & currentOrientation, int iNode, int iTurbGlob);
-  void getForce(std::vector<double> & force, int iNode, int iTurbGlob);
+  //! Get the coordinates of force node number 'iNode' for turbine number 'iTurbGlob'
+  void getForceNodeCoordinates(std::vector<double> & currentCoords, int iNode, int iTurbGlob, fast::timeStep t);
+  //! Get the tensor orientation of force node number 'iNode' for turbine number 'iTurbGlob'
+  void getForceNodeOrientation(std::vector<double> & currentOrientation, int iNode, int iTurbGlob, fast::timeStep t);
+  //! Get the actuator force at force node 'iNode' for turbine number 'iTurbGlob'
+  void getForce(std::vector<double> & force, int iNode, int iTurbGlob, fast::timeStep t);
+  //! Get the chord at force node 'iNode' for turbine number 'iTurbGlob'
   double getChord(int iNode, int iTurbGlob);
 
+  //! Get the starting time step of the simulation
   int get_ntStart() { return ntStart; }
+  //! Return a boolean flag whether the simulation is dryRun
   bool isDryRun() { return dryRun; }
+  //! Return a boolean flag whether the simulation is debug
   bool isDebug() { return debug; }
+  //! Get an enum of type 'simStartType' indicating the start type of the simulation
   simStartType get_simStartType() { return simStart; }
+  //! Is the simulation time zero right now?
   bool isTimeZero() { return timeZero; }
-  int get_procNo(int iTurbGlob) { return turbineMapGlobToProc[iTurbGlob] ; } // Get processor number of a turbine with global id 'iTurbGlob'
+  //! Get the global MPI rank of the processor containing turbine number 'iTurbGlob'
+  int get_procNo(int iTurbGlob) { return turbineMapGlobToProc[iTurbGlob] ; }
+  //! Get the local turbine number of the turbine number 'iTurbGlob'
   int get_localTurbNo(int iTurbGlob) { return reverseTurbineMapProcToGlob[iTurbGlob]; }
+  //! Get the total number of turbines in the simulation
   int get_nTurbinesGlob() { return nTurbinesGlob; } 
 
+  //! Get the nacelle drag coefficient of turbine number 'iTurbGlob'
   float get_nacelleArea(int iTurbGlob) { return get_nacelleAreaLoc(get_localTurbNo(iTurbGlob)); }
```

Comment:
Is this safe to call on every MPI rank?

Reply:
Ah... no. I realize this is true of many functions. I've added comments in the documentation above each function to indicate whether a particular function MUST be called from every MPI rank, is safe to be called from every MPI rank, or should be called only from the processor containing the turbine. Implemented in commit 9f34f9eb119d38dc0c78bfe58fdd23d084d02430.